src/main/scala/org/constellation/lb/Manager.scala

```diff
@@ -55,179 +59,235 @@ class Manager(init: NonEmptyList[Addr], config: LoadbalancerConfig)(
-      .flatTap(_ => logger.info("Scheduling next cluster status update round."))
-      .flatMap(_ => IO.sleep(60 seconds))
-      .flatMap(_ => manager)
-  }
+private val settingsRoutes = HttpRoutes.of[IO] {
+  case POST -> Root / "maintenance" =>
+    enableMaintenanceMode.flatMap(_ => Ok())
+  case DELETE -> Root / "maintenance" =>
+    disableMaintenanceMode.flatMap(_ => Ok())
+}
+
+
+def buildClusterStatus(init: NonEmptyMap[Addr, Option[List[Info]]]): IO[(Set[Addr], Set[Addr])] = {
+  val tresholdLevel = init.keys.size / 2
+
+  def isActive(addr: Addr, proof: List[Info]) =
+    init(addr).nonEmpty && proof.count(_.state == NodeState.Ready) > tresholdLevel
 
-  def run: IO[ExitCode] =
-    http
-      .use { client =>
-        NonEmptyList.of(settingsServer, lb.server, manager(client)).parSequence
+  IO {
+    val (active, other) = init.toNel
+      .collect {
+        case (_, Some(el)) => el
       }
-      .as(ExitCode.Success)
-
-  def isUnderMaintenance: IO[Boolean] = underMaintenance.get
-
-  def enableMaintenanceMode: IO[Unit] =
-    for {
-      _ <- underMaintenance.modify(_ => (true, ()))
-      _ <- logger.info("Enabled maintenance mode.")
-      _ <- lb.enableRedirectingToMaintenance
-    } yield ()
-
-  def disableMaintenanceMode: IO[Unit] =
-    for {
-      _ <- lb.reset
-      _ <- reset
-      _ <- underMaintenance.modify(_ => (false, ()))
-      _ <- logger.info(
-        "Disabled maintenance mode. Load balancer will be available after next cluster status update round."
-      )
-    } yield ()
-
-  private val utilsHealth = HttpRoutes.of[IO] {
-    case GET -> Root / "health" => Ok()
-    case GET -> Root / "health" / addrStr =>
-      Addr
-        .unapply(addrStr)
-        .map { addr =>
-          hostsRef.get.flatMap(
-            addrs =>
-              addrs(addr)
-                .map {
-                  case None    => NoContent()
-                  case Some(_) => Ok()
-                }
-                .getOrElse(NotFound())
-          )
-        }
-        .getOrElse(BadRequest())
+      .flatten
+      .groupBy(n => Addr(n.ip, n.publicPort))
+      .toList
+      .partition {
+        case (addr: Addr, proof: List[Info]) => isActive(addr, proof)
+      }
+
+    active.map(_._1).toSet -> other.map(_._1).toSet
+  }.flatTap {
+    case (active, other) =>
+      logger.info(s"As a result of status analysis we have ${active} hosts and ${other} not ready")
   }
+}
 
-  private val settingsServer: IO[Unit] =
-    logger
-      .info(s"Setup Settings instance on ${config.`if`}:${config.settingsPort}")
-      .flatMap(
-        _ =>
-          http.use(
-            _ =>
-              BlazeBuilder[IO]
-                .bindHttp(config.settingsPort, config.`if`)
-                .mountService(utilsHealth, "/utils")
-                .mountService(settingsRoutes, "/settings")
-                .serve
-                .compile
-                .drain
+def getClusterInfo(
+                    hosts: NonEmptySet[Addr]
```

Comment:
I guess you need to configure the code formatter properly. The project uses https://scalameta.org/scalafmt/

Reply:
dang, intelliJ is not picking it pu :( will fix it

Reply:
ran scalafmt from command line :)

Reply:
@gclaramunt looks like the formatter that you've used made much more changes that ones required for your change :| is it possible that you commit reformatted code (with no more logic changes) as a separate commit, and your changes as a separate commit? right now it's hard to distinguish what is the effect of the formatter and what is really a change needed here.

Reply:
I've reverted back the format changes, even if there are some now, they're are related to the logic change (minus removing some unused imports)

Reply:
<img width="683" alt="image" src="https://github.com/user-attachments/assets/9b7ffe11-b236-4767-836f-a5329021268f">

Just make sure that you have it set to Scalafmt instead of IntelliJ @gclaramunt